mParticle/mparticle-roku-sdk
# feat: add package.json

## Summary
- In the projects we are working on, all of our third-party code is installed using `npm` and imported into the project using several automated configurable scripts. The reason for this is to ensure proper governance and security/audit procedures.
- We need to install mParticle SDK in one of our projects, but we are unable to automate installation using `npm`, because it lacks a `package.json` file (and adding the files manually is not an option due to the reason mentioned above).
- This PR adds a small `package.json` file, so anyone is able to install the dependency, like shown in the example below:

```json
{
    "name": "my-custom-roku-app",
    ...
    "dependencies": {
        ...
        "mparticle": "mParticle/mparticle-roku-sdk.git#v2.1.14"
    }
}
```

## Testing Plan
- Generate a new npm project using `mkdir my-custom-roku-app && cd my-custom-roku-app && npm init -y`
- Install using the source repo: `npm i mParticle/mparticle-roku-sdk#v2.1.14`
  - **This should fail,** with an error similar to:
  - `npm ERR! enoent ENOENT: no such file or directory, open '/Users/lse16/.npm/_cacache/tmp/git-cloneNClu2N/package.json'`
- Install using the forked repo: `npm i luis-soares-sky/mparticle-roku-sdk#feat/add-package-json`
  - **This should work,** with a new entry added to the project's dependencies:
  - `"mparticle-roku-sdk": "github:luis-soares-sky/mparticle-roku-sdk#feat/add-package-json"`
